identity-mgt/code-migration/org.wso2.carbon.identity.securitycode.migration/src/main/java/org/wso2/carbon/identity/securitycode/migration/internal/IdentityCodeMigrationServiceComponent.java

```diff
@@ -0,0 +1,148 @@
+/*
+ * Copyright (c) 2018, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.wso2.carbon.identity.securitycode.migration.internal;
+
+import org.apache.commons.logging.Log;
+import org.apache.commons.logging.LogFactory;
+import org.osgi.service.component.ComponentContext;
+import org.osgi.service.component.annotations.Activate;
+import org.osgi.service.component.annotations.Component;
+import org.osgi.service.component.annotations.Reference;
+import org.osgi.service.component.annotations.ReferenceCardinality;
+import org.osgi.service.component.annotations.ReferencePolicy;
+import org.wso2.carbon.identity.core.util.IdentityCoreInitializedEvent;
+import org.wso2.carbon.identity.securitycode.migration.service.SecurityCodeMigrationService;
+import org.wso2.carbon.registry.core.service.RegistryService;
+import org.wso2.carbon.user.core.service.RealmService;
+
+import static org.apache.commons.lang.StringUtils.isBlank;
+import static org.apache.commons.lang.StringUtils.isNotBlank;
+import static org.wso2.carbon.identity.securitycode.migration.MigrationConstants.SYSTEM_PROPERTY_MIGRATE;
+import static org.wso2.carbon.identity.securitycode.migration.MigrationConstants.SYSTEM_PROPERTY_TENANTS;
+import static org.wso2.carbon.identity.securitycode.migration.MigrationConstants.TENANT_DELIMITER;
+
+@Component(
+        name = "IdentityCodeMigrationServiceComponent",
+        immediate = true
+)
+public class IdentityCodeMigrationServiceComponent {
+
+    private static Log log = LogFactory.getLog(IdentityCodeMigrationServiceComponent.class);
+
+    @Activate
+    protected void activate(ComponentContext context) {
+
+        if (log.isDebugEnabled()) {
+            log.debug("Security code migration bundle is activated");
+        }
+
+        SecurityCodeMigrationService securityCodeMigrationService = new SecurityCodeMigrationService();
+        String migrateCode = System.getProperty(SYSTEM_PROPERTY_MIGRATE);
+        String tenantToMigrate = System.getProperty(SYSTEM_PROPERTY_TENANTS);
```

Comment:
Possible improvement to consider:
Provide system property for a tenant range.
When there are around 3000+ tenants it becomes difficult to test migrations as it will take some time.If we can provide a property for set of tenant range, they can migrate chunk wise.

Reply:
Added below